### PR TITLE
Reader | Receipt Date and Document Type become buttons

### PR DESCRIPTION
**Pull request by mmlumba**
Converts the fields "Receipt Date" and "Document Type" to buttons so they're accessible to keyboards.
Connects #1847 
document-list-buttons

### CODE REVIEW

File: client/app/reader/DocumentsTable.jsx [hunk 1]
```diff
@@ -251,8 +251,7 @@ class DocumentsTable extends React.Component {
       {
         cellClass: 'categories-column',
         header: <div
-          id="categories-header"
-          className="document-list-header-categories">
```
Comment (mmlumba) [inline client/app/reader/DocumentsTable.jsx]: Class was unused so I removed it

File: client/app/reader/DocumentsTable.jsx [hunk 2]
```diff
@@ -277,22 +276,21 @@ class DocumentsTable extends React.Component {
       },
       {
         cellClass: 'receipt-date-column',
-        header: <div
+        header: <button
           id="receipt-date-header"
-          className="document-list-header-recepit-date"
```
Comment (mmlumba) [inline client/app/reader/DocumentsTable.jsx]: Class was unused so I removed it

File: client/app/reader/DocumentsTable.jsx [hunk 3]
```diff
@@ -277,22 +276,21 @@ class DocumentsTable extends React.Component {
       },
       {
         cellClass: 'receipt-date-column',
-        header: <div
+        header: <button
```
Comment (mdbenjam) [inline client/app/reader/DocumentsTable.jsx]: Can we use the `Button` component?

File: spec/feature/reader_spec.rb [hunk 4]
```diff
@@ -221,6 +221,15 @@ def expect_dropdown_filter_to_be_visible
       expect_dropdown_filter_to_be_visible
 
       expect(find("#procedural", visible: false).checked?).to be false
+
+      find("#receipt-date-header").send_keys :enter
```
Comment (mdbenjam) [inline spec/feature/reader_spec.rb]: I actually don't think we need these tests. We test the sorting in a javascript test, and I don't think we need a check here that a button is accessible.

**Pull request merged**